hanson_at_fb2_iterative.py

```diff
@@ -0,0 +1,30 @@
+# given a list of unqiue, integers, return all possible subsets
+# input: [1, 2, 3]
+# output: [[1], [2], [3], [1, 2], [1, 3], [2, 3], [1,2,3] []]
+
+# base case, input [], output []
+# input [1], output [], [1]
+# input [2], output [], [2], [1], [1, 2]
+# every time you add the new number to previously existing elements
+
+def subset_list (int_list):
+	all_subsets = [[]]
+
+	for input_integer in int_list:
+		subset = []
+		for known_subsets in all_subsets:
```

Comment:
This is good! To make this even more concise, can we rewrite this with a `map`? You can either define a new function, or maybe even just use a `lambda` in this case.

Hint: The map will remove the need for a `subset` variable